docs/testing.rst

```diff
@@ -1,71 +1,321 @@
+
+  * If ``data/`` directory is present and does not contain any data files, a
+    new empty database is used in the tests;
+
+  * If ``data/`` directory is absent, the suite will use a database populated
+    with test data from the ``CITE`` suite.
+
+* An optional ``get/requests.txt`` file that holds request parameters used for
+  making HTTP GET requests.
+
+  Each line in the file must be formatted with the following scheme:
+
+      test_id,request_query_string
+
+  For example:
+
+    TestGetCapabilities,service=CSW&version=2.0.2&request=GetCapabilities
+
+  When tests are run, the *test_id* is used for naming each test and for
+  finding the expected result.
+
+* An optional ``post/`` directory that holds ``.xml`` files used for making
+  HTTP POST requests
+
+
+Test identifiers
+^^^^^^^^^^^^^^^^
+
+Each test has an identifier that is built using the following rule:
+
+    <test_function>[<suite_name>_<http_method>_<test_name>]
+
+For example:
+
+    test_suites[default_post_GetRecords-end]
+
+
+Running tests
+-------------
+
+Since pycsw uses `pytest`_, tests are run with the ``py.test`` runner. A basic
+test run can be made with:
+
+.. code:: bash
+
+   py.test
+
+This command will run all tests and report on the number of successes, failures
+and also the time it took to run them. The `py.test` command accepts several
+additional parameters that can be used in order to customize the execution of
+tests. Look into `pytest's invocation documentation`_ for a more complete
+description. You can also get a description of the available parameters by
+running:
+
+.. code:: bash
+
+   py.test --help
+
+
+
+Running specific suites and test cases
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Use the ``-k <name_expression>`` flag to select which tests to run. Since each
+test's name includes the suite name, http method and an identifier for the
+test, it is easy to run only certain tests.
+
+.. code:: bash
+
+   py.test -k "apiso and GetRecords"  # run only tests from the apiso suite that have GetRecords in their name
+   py.test -k "post and GetRecords"  # run only tests that use HTTP POST and GetRecords in their name
+   py.test -k "not harvesting"  # run all tests except those from the harvesting suite
 
-.. _tests:
+
+Exiting fast
+^^^^^^^^^^^^
+
+The ``--exitfirst`` (or ``-x``) flag can be used to stop the test runner
+immediately as soon as a test case fails.
+
+.. code:: bash
+
+   py.test --exitfirst
+
+
+Seeing more output
+^^^^^^^^^^^^^^^^^^
+
+There are three main ways to get more output from running tests:
+
+* The ``--verbose`` (or ``-v``) flag;
+
+* The ``--capture=no`` flag - Messages sent to stdout by a test are not
+  suppressed;
+
+* The ``--pycsw-loglevel`` flag - Sets the log level of the pycsw instance
+  under test. Set this value to ``debug`` in order to see all debug messages
+  sent by pycsw while processing a request.
+
+
+.. code:: bash
+
+   py.test --verbose
+   py.test --pycsw-loglevel=debug
```

Comment:
Yes, please see [here](https://github.com/ricardogsilva/pycsw/blob/issue%23428/tests/functionaltests/conftest.py#L190).

It actually sets `server.loglevel` to the value supplied by the `--pycsw-loglevel` py.test command-line argument and `server.logfile` to an empty string. This makes the logs go out to stdout and makes them visible when running tests